classes/Cart.php

```diff
@@ -755,6 +755,15 @@ public function getProducts($refresh = false, $id_product = false, $id_country =
                     $reduction_type_row = ['reduction_type' => 0];
                 }
 
+                if (Pack::isPack($product['id_product'])) {
+                    /*
+                    * In case of packs, we need to properly calculate the quantity in stock. We can't just use the quantity from the query, because stocks can be set to use the quantity of the products in them. The quantity in stock_available then has no meaning and could be always zero.
+                    *
+                    * When calling Pack::getQuantity here, you MUST use null for $cart parameter. Otherwise it will subtract the quantity that is already in the cart. Basically resulting in a nonsense - half of quantity you have. We need the REAL quantity.
+                    */
+                    $product['quantity_available'] = Pack::getQuantity((int) $product['id_product'], (int) $product['id_product_attribute'], true, null, (int) $product['id_customization']);
```

Comment:
```suggestion
                    $products[$key]['quantity_available'] = Pack::getQuantity((int) $product['id_product'], (int) $product['id_product_attribute'], true, null, (int) $product['id_customization']);
```

otherwise it won't work since `$product` is not passed by a reference

Reply:
Doing so doesn't work because `$products[$key]['quantity_available']` is overwritten by the statement `$products[$key] = array_merge($product, $reduction_type_row);`
In order to use your suggestion I would have to move everything after `$products[$key] = array_merge($product, $reduction_type_row);`
so we would have:

```
                $products[$key] = array_merge($product, $reduction_type_row);
                if (Pack::isPack($product['id_product'])) {
                    /*
                    * In case of packs, we need to properly calculate the quantity in stock. We can't just use the quantity from the query, because stocks can be set to use the quantity of the products in them. The quantity in stock_available then has no meaning and could be always zero.
                    *
                    * When calling Pack::getQuantity here, you MUST use null for $cart parameter. Otherwise it will subtract the quantity that is already in the cart. Basically resulting in a nonsense - half of quantity you have. We need the REAL quantity.
                    */
                    $products[$key]['quantity_available'] = Pack::getQuantity((int) $product['id_product'], (int) $product['id_product_attribute'], true, null, (int) $product['id_customization']);
                }
```

I'm not assuming what I said, I've done real tests.